### PR TITLE
[availability] Adjust design to set active field on each block

### PR DESCRIPTION
**Pull request by fxfactorial**
...after convo with RJ, I had to pull over at a starbucks in millbrae to bang out this much more elegant solution.

### CODE REVIEW

File: internal/chain/engine.go [hunk 1]
```diff
@@ -292,6 +296,7 @@ func (e *engineImpl) Finalize(
 				)
 				state.AddBalance(delegation.DelegatorAddress, totalWithdraw)
 			}
+			wrapper.LastEpochInCommittee = newShardState.Epoch
```
Comment (rlan35) [inline internal/chain/engine.go]: This is not the right way. LastEpochInCommittee should only be set if the validator is in the shardState.

File: staking/availability/measure.go [hunk 2]
```diff
@@ -186,14 +186,15 @@ type Reader interface {
...
 
-// SetInactiveUnavailableValidators sets the validator to
+// ComputeAndRecord sets the validator to
 // inactive and thereby keeping it out of
 // consideration in the pool of validators for
 // whenever committee selection happens in future, the
 // signing threshold is 66%
-func SetInactiveUnavailableValidators(
-	bc Reader, state *state.DB, addrs []common.Address,
+func ComputeAndRecord(
```
Comment (rlan35) [inline staking/availability/measure.go]: Why not move the logic here into the IncrementValidatorSigningCounts? Then you don't need to do Read and UpdateValidatorWrapper twice.

File: internal/chain/engine.go (hunk 1 above)
Comment (fxfactorial) [reply internal/chain/engine.go]: Hmm, that’s how it was before I merged big slashing PR

File: staking/availability/measure.go (hunk 2 above)
Comment (fxfactorial) [reply staking/availability/measure.go]: Thought about it but it’s a separation of concerns. But sure can fold them both together

File: internal/chain/engine.go (hunk 1 above)
Comment (rlan35) [reply internal/chain/engine.go]: No, it's not, just see the deleted code on the left side, it was ranging over shardState.validators. Now it's ranging over all validators.

Comment (fxfactorial) [reply internal/chain/engine.go]: You're right, just looked a bit closer on older version code: 

```
 git show 801b4b83e277bfc50600d448fed3774cd4864650~1:internal/chain/engine.go
```

**Pull request merged**